spidermon/contrib/stats/statscollectors.py

```diff
@@ -0,0 +1,35 @@
+import os
+import pickle
+from collections import deque
+
+from scrapy.statscollectors import StatsCollector
+from scrapy.utils.project import data_path
+
+
+class LocalStorageStatsHistoryCollector(StatsCollector):
+    def __init__(self, crawler):
+        super(LocalStorageStatsHistoryCollector, self).__init__(crawler)
+
+        self._statsdir = data_path("stats", createdir=True)
+
+    def open_spider(self, spider):
+        stats_location = os.path.join(
+            self._statsdir, "{}_stats_history".format(spider.name)
+        )
+        if os.path.isfile(stats_location):
+            with open(stats_location, "rb") as stats_file:
+                _stats_history = pickle.load(stats_file)
+        else:
+            max_stored_stats = spider.crawler.settings.getint(
+                "SPIDERMON_MAX_STORED_STATS", default=100
+            )
+            _stats_history = deque([], maxlen=max_stored_stats)
+        spider.stats_history = _stats_history
```

Comment:
Regarding the preceding if-else statement, it looks to me like `SPIDERMON_MAX_STORED_STATS` is currently ignored after the first time a spider is run. Shouldn’t the `maxlen` of the `deque` loaded with `pickle` be adjusted according to the `SPIDERMON_MAX_STORED_STATS` setting as well?

If so, maybe adding a test that specifically ensures that new runs adjust the `deque.maxlen` value accordingly could prove useful.

Reply:
It makes sense. I updated the code and the tests for this scenario.